src/XrdPfc/XrdPfcFile.cc

```diff
@@ -820,13 +832,26 @@ int File::ReadOpusCoalescere(IO *io, const XrdOucIOVec *readV, int readVnum,
             {
                TRACEF(DumpXL, tpfx << "direct block " << block_idx << ", blk_off " << blk_off << ", size " << size);
 
-               if (lbe == LB_direct)
-                  iovec_direct.back().size += size;
-               else
-                  iovec_direct.push_back( { block_idx * m_block_size + blk_off, size, 0, iUserBuff + off } );
                iovec_direct_total += size;
                read_req->m_direct_done = false;
 
+               // Make sure we do not issue a ReadV with chunk size above XrdProto::maxRVdsz.
+               // Number of actual ReadVs issued so as to not exceed the XrdProto::maxRvecsz limit
+               // is determined in the RequestBlocksDirect().
+               if (lbe == LB_direct && iovec_direct.back().size + size <= XrdProto::maxRVdsz) {
+                  iovec_direct.back().size += size;
+               } else {
+                  long long  in_offset = block_idx * m_block_size + blk_off;
+                  char      *out_pos   = iUserBuff + off;
+                  while (size > XrdProto::maxRVdsz) {
+                     iovec_direct.push_back( { in_offset, XrdProto::maxRVdsz, 0, out_pos } );
+                     in_offset += XrdProto::maxRVdsz;
+                     out_pos   += XrdProto::maxRVdsz;
+                     size      -= XrdProto::maxRVdsz;
+                  }
+                  iovec_direct.push_back( { in_offset, size, 0, out_pos } );
```

Comment:
Same here
```suggestion
                  iovec_direct.emplace_back(in_offset, size, 0, out_pos);
```

Reply:
For this to work I have to add ctors to XrdOucIOVec:
```
       XrdOucIOVec() = default;
       XrdOucIOVec(long long offs, int sz, int inf, char *buff) :
         offset(offs), size(sz), info(inf), data(buff) {}
```
here: https://github.com/xrootd/xrootd/blob/master/src/XrdOuc/XrdOucIOVec.hh#L46

Do you want me to go forward with that?

Reply:
Presumably the ctors will do nothing, right?

Reply:
Yes, they do nothing ... and they keep the automatic move ctor and operator=, the copy ctor and operator=.

Reply:
If the replacement I suggested doesn't work out of the box, then consider my suggestions optional. I wonder why the initializer list works then, need to take a look at this later.

Reply:
I've been wondering about this myself ... and concluded that it needs the argument-ed constructor, as done above. Initializer list was automatically promoted to move-constructor. Initializer lists are very eager to become types, too eager, I hear people say ;)
```
void push_back( T&& value );
// vs.
template< class... Args >
void /* reference from c++17 on */ emplace_back( Args&&... args );
```